app/controllers/api/metrics_controller.rb

```diff
@@ -10,29 +10,26 @@ def show
                  .order('dimensions_dates.date asc')
                  .group('dimensions_dates.date')
                  .sum(metric)
-  end
-
-private
-
-  METRIC_WHITELIST = %w[pageviews unique_pageviews].freeze
 
-  def content_id
-    @content_id ||= params[:content_id]
+    @metric_params = metric_params
   end
 
-  def from
-    @from ||= params[:from]
-  end
+private
 
-  def to
-    @to ||= params[:to]
-  end
+  delegate :from, :to, :metric, :content_id, to: :metric_params
 
-  def metric
-    @metric ||= params[:metric]
+  def metric_params
+    @metric_params ||= Api::Metric.new(params)
   end
 
-  def validate_metric!
-    head :bad_request unless Facts::Metric.valid_metric? metric
+  def validate_params!
+    unless metric_params.valid?
+      response = {
+        type: "https://content-performance-api.publishing.service.gov.uk/errors/#validation-error",
```

Comment:
Shouldn't we be using `Plek`? 
Can you explain a bit more in the commit notes the details around the name: `content-performance-api`?